app/models/concerns/key_pair_compute_resource.rb

```diff
@@ -5,7 +5,7 @@ module KeyPairComputeResource
     prepend KeyPairCapabilities
     has_one :key_pair, :foreign_key => :compute_resource_id, :dependent => :destroy
     after_create :setup_key_pair
-    after_destroy :destroy_key_pair
+    after_destroy :delete_key_from_resource
```

Comment:
This method may return false on failure (if `key.destroy` returns false, and possibly on `rescue`), halting the callback, which is different behaviour to the commit prior to 59e7136. Should it remain wrapped to ignore failures?